# Atividade T1.2 (Correto)

Discretizar e resolver a equação $\frac{\partial^{2}}{\partial x^{2}}+\frac{\partial^{2}}{\partial y^{2}}=0$ no quadrado [0,1]×[0,1] com condição de contorno dada por 

$$
\begin{aligned}
T(x,0)=100\sqrt{x}, T(x,1)=100(\sqrt{x}−1), para 0≤x≤1;\\
T(0,y)=−100\sqrt{y}, T(1,y)=100(1−\sqrt{y}), para 0≤y≤1.
\end{aligned}
$$

Divida o intervalo [0,1] em 3 subintervalos de mesmo comprimento e considere a discretização em que 

T1 é a temperatura no ponto $[\frac{1}{3},\frac{2}{3}];$

T2 é a temperatura no ponto $[\frac{2}{3},\frac{2}{3}];$

T3 é a temperatura no ponto $[\frac{1}{3},\frac{1}{3}];$

T4 é a temperatura no ponto $[\frac{2}{3},\frac{1}{3}];$

Resolva o sistema pelo Método do Escalonamento, fazendo todas as contas com exatamente 5 algarismos significativos. 

### Resolução

Podemos calcular T1, T2, T3 e T4, como a discretização da temperatura no ponto dado:

$$
\begin{aligned}
T_1= \frac{T(\frac{1}{3}, 1) + T_2 + T_3 + T(0, \frac{2}{3})}{4}\\
T_2= \frac{T(\frac{2}{3}, 1) + T_1 + T_4 + T(1, \frac{2}{3})}{4}\\
T_3= \frac{T(0, \frac{1}{3}) + T_1 + T_4 + T(\frac{1}{3}, 0)}{4}\\
T_4= \frac{T(\frac{2}{3}, 0) + T_3 + T_2 + T(1, \frac{1}{3})}{4}\\
\end{aligned}
$$



In [2]:
# Funções de temperatura
function [] = Tx0(x) 
  100*(sqrt(x))
end
function [] = Tx1(x) 
  100*(sqrt(x) - 1)
end
function [] = T0y(y) 
  (-100)*(sqrt(y))
end
function [] = T1y(y) 
  100*(1 - sqrt(y))
end

Tx1(1/3)
T0y(2/3)
Tx1(2/3)
T1y(2/3)
T0y(1/3)
Tx0(1/3)
Tx0(2/3)
T1y(1/3)

ans = -42.265
ans = -81.650
ans = -18.350
ans = 18.350
ans = -57.735
ans = 57.735
ans = 81.650
ans = 42.265



Sabemos que:

$$
\begin{aligned}
T(\frac{1}{3}, 1) = -42.265, \hspace{0.5cm}
T(0, \frac{2}{3}) = -81.650, \hspace{0.5cm}
T(\frac{2}{3}, 1) = -18.350\\
T(1, \frac{2}{3}) = 18.350, \hspace{0.5cm}
T(0, \frac{1}{3}) = -57.735, \hspace{0.5cm}
T(\frac{1}{3}, 0) = 57.735\\
T(\frac{2}{3}, 0) = 81.650, \hspace{0.5cm}
T(1, \frac{1}{3}) = 42.265
\end{aligned}
$$

Logo, 

$$
\begin{aligned}
T_1= \frac{-123.92 + T_2 + T_3}{4}, \hspace{0.5cm}
T_2= \frac{T_1 + T_4}{4}\\
T_3= \frac{T_1 + T_4}{4}, \hspace{0.5cm}
T_4= \frac{123.92 + T_3 + T_2}{4}\\
\Downarrow\\
T_1 - \frac{T_2}{4} - \frac{T_3}{4} = -\frac{123.92}{4}, \hspace{0.5cm}
T_2 - \frac{T_1}{4}- \frac{T_4}{4} = 0\\ 
T_3 - \frac{T_1}{4}- \frac{T_4}{4} = 0, \hspace{0.5cm}
T_4 - \frac{T_2}{4} - \frac{T_3}{4} = +\frac{123.92}{4}
\end{aligned}
$$

Que pode ser escrito como:

$$
\begin{bmatrix}
1 & -0.2500 & -0.2500 & 0 & -30.980\\
-0.2500 & 1 & 0 & -0.2500 & 0\\
-0.2500 & 0 & 1 & -0.2500 & 0\\
0 & -0.2500 & -0.2500 & 1 & 30.980
\end{bmatrix}
$$

In [4]:
pkg install "https://downloads.sourceforge.net/project/octave/Octave%20Forge%20Packages/Individual%20Package%20Releases/symbolic-3.0.1.tar.gz"
pkg load symbolic
setenv Python "/usr/bin/python3"

For information about changes from previous versions of the symbolic package, run 'news symbolic'.


In [23]:
d = digits(5)
A = vpa([1 , -0.25000 , -0.25000 , 0;
-0.25000 , 1 , 0 , -0.25000;
-0.25000 , 0 , 1 , -0.25000;
0 , -0.25000 , -0.25000 , 1]);
b= vpa([-30.980; 0; 0; 30.980]);

[x,A,b] = gauss(A,b);
x

d = 5
x = (sym 4×1 matrix)

  ⎡-30.98⎤
  ⎢      ⎥
  ⎢  0   ⎥
  ⎢      ⎥
  ⎢  0   ⎥
  ⎢      ⎥
  ⎣30.98 ⎦



Usando o escalonador com controle de algarismos significativos temos:

T1 = -30,98 (V)
T2 = 0,00032 (v)
T3 = 0,00042857 (v)
T4 = 30,98 (v)
